WooCommerce/src/main/kotlin/com/woocommerce/android/util/CouponUtils.kt

```
@@ -80,6 +80,36 @@ class CouponUtils @Inject constructor(
         }
     }
 
+    /*
+    - If all products are included: "Apply 15% off to all products with the promo code ABCDE"
+    - If only some products: "Apply 15% off to select products with the promo code ABCDE"
+     */
+    fun formatSharingMessage(
+        amount: BigDecimal?,
+        currencyCode: String?,
+        couponCode: String,
+        includedProducts: Int,
+        excludedProducts: Int
+    ): String {
+        return if (amount != null && currencyCode != null) {
+            if (includedProducts == 0 && excludedProducts == 0) {
+                resourceProvider.getString(
+                    R.string.coupon_details_share_coupon_all,
+                    formatCurrency(amount, currencyCode),
+                    couponCode
+                )
+            } else {
+                resourceProvider.getString(
+                    R.string.coupon_details_share_coupon_some,
+                    formatCurrency(amount, currencyCode),
+                    couponCode
+                )
+            }
+        } else {
+            ""
```

What happens in this case? Will the empty string be used?

---

Yes. I'm a bit doubtful about this as well. What do you think will be ideal for error handling here? I feel if the data is incomplete (amount is null or currency is null) then we shouldn't allow it to be shared, since it will be confusing to recipients of this shared message. My thinking is that if it's empty, at least the message won't be misleading.

---

A good rule is that you want to fail as soon as possible. If you know that a certain value is required and it doesn't make sense to show a screen without it, you shouldn't proceed with some dummy value (like an empty string), because it's not useful to the user. Also, you want to identify a problem if there is an error coming from the endpoint. So here, we might want to use a `null` value instead and display an error if the value is missing and log it.